### PR TITLE
Check for Gemfile before running pod commands

### PR DESCRIPTION
**Pull request by birmacher**

### CODE REVIEW

**Comment by birmacher** [comment]
Also solves issue #435

**Comment by KrauseFx** [comment]
I don't think running `bundle install` because there is a `Gemfile` is a good idea, as people without `rvm` will run into permission issues. Instead `bundle install` should be its own `fastlane` action.

So, just pre-fixing the `pod install` with `bundle exec` when there is a `Gemfile` should be good :+1:

**Comment by birmacher** [comment]
would you like to also add it to the `FastfileTemplate` and to the setup process?

File: lib/fastlane/actions/bundle_install.rb [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+          cmd << "--with #{params[:with]}" if params[:with]
+
+          Actions.sh(cmd.join(' '))
+        else
+          Helper.log.info "No Gemfile found"
+        end
+      end
+
+      def self.description
+        'This action runs `bundle install` if it founds the Gemfile'
```
Comment (KrauseFx) [inline lib/fastlane/actions/bundle_install.rb]: `if it founds the Gemfile.` probably not necessary?

**Comment by KrauseFx** [comment]
This looks great, thanks @birmacher :+1:

**Comment by sethhoward** [comment]
Recent update now seems to for 'bundle exec pod install'. This is undesirable as the Gemfile in our project is strictly for Calabash use. Running cocoapods as an action is no longer available.

This also makes the documentation for the cocoapods action incorrect currently says:  Runs `pod install` for the project   
Where in reality it seems to be running: bundle exec pod install

**Comment by KrauseFx** [comment]
@sethhoward so what do you suggest? Why don't you add `cocoapods` to your `Gemfile`? Should there be an option like `skip_bundle_exec` in the `cocoapods` action?

**Comment by sethhoward** [comment]
Good question. Our Gemfile is maintained by our QA team and we don't bother much with it as a platform. Adding cocoapods to the Gemfile created several other issues before I posted so we turned our back on it.

skip_bundle_exec would be a totally acceptable solution and was the second thing I looked for in the documentation before posting.

**Comment by KrauseFx** [comment]
@sethhoward I think running `bundle exec` is the right way to do this. I'd suggest you add the `fastlane` dependency to your `Fastfile` as recommended.

**Comment by KrauseFx** [comment]
@sethhoward this will be in the next update https://github.com/KrauseFx/fastlane/pull/512

**Comment by KrauseFx** [comment]
The version is now live, thanks again :+1:

**Pull request merged**